Samples/DrTestActionSampleVM/DrTestActionSampleVM/WrapperDrVM.cs

```diff
@@ -258,6 +271,339 @@ protected virtual VirtualMachine GetVirtualMachine(string vmName)
+            dcmor.Type = networkSystem.MoRef.Type;
+            dcmor.Value = networkSystem.MoRef.Value;
+
+            HostNetworkSystem _switch = new HostNetworkSystem(vClient, dcmor);
+            _switch.RemoveVirtualSwitch(switchName);
+
+        }
+
+
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="vmName"></param>
+        /// <param name="portGroup"></param>
+        internal virtual void ChangeNicPortGroup(string vmName, string portGroup)
+        {
+
+            var vm = GetVirtualMachine(vmName);
+            if (vm == null) throw new VMDoesntExistExeption(vmName);
+            VirtualMachineConfigSpec vmConfigSpec = new VirtualMachineConfigSpec();
+            VirtualDeviceConfigSpec[] nicSpec = HelperGetNICDeviceConfigSpec(vm, portGroup);
+            vmConfigSpec.DeviceChange = nicSpec;
+            vm.ReconfigVM(vmConfigSpec);
+        }
+
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="vm"></param>
+        /// <param name="pgname"></param>
+        /// <returns></returns>
+        private static VirtualDeviceConfigSpec[] HelperGetNICDeviceConfigSpec(VirtualMachine vm, String pgname)
+        {
+
+
+
+
+
+            List<VirtualDeviceConfigSpec> updates = new List<VirtualDeviceConfigSpec>();
+            VirtualMachineConfigInfo vmConfigInfo = vm.Config;
+            VirtualDevice[] vds = vm.Config.Hardware.Device;
+
+            for (int i = 0; i < vds.Length; i++)
+            {
+                if (vds[i] is VirtualEthernetCard)
+                {
+                    VirtualDeviceConfigSpec nicSpec = new VirtualDeviceConfigSpec();
+                    nicSpec.Operation = VirtualDeviceConfigSpecOperation.edit;
+                    VirtualEthernetCardNetworkBackingInfo oldbi = (VirtualEthernetCardNetworkBackingInfo)vds[i].Backing;
+                    VirtualEthernetCardNetworkBackingInfo bi = new VirtualEthernetCardNetworkBackingInfo();
+                    bi.DeviceName = pgname;
+                    vds[i].Backing = bi;
+                    nicSpec.Device = vds[i];
+                    updates.Add(nicSpec);
+                }
+            }
+
+            VirtualDeviceConfigSpec[] ret = new VirtualDeviceConfigSpec[updates.Count];
+            int x = 0;
+            foreach (var test in updates)
+            {
+                ret[x] = test;
+                x++;
+            }
+            return ret;
+
+        }
+
+
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="vmName"></param>
+        /// <param name="login"></param>
+        /// <param name="password"></param>
+        /// <param name="application"></param>
+        /// <param name="retry"></param>
+        /// <param name="timeout"></param>
+        internal virtual void CheckVMState(string vmName, string login, string password, string application, string retry, string timeout)
```

Comment:
should add summary. CheckVMState(string vmName, string guestLogin, string guestPwd, string applicationName, string attempts, string attemptsTimeOut)